core/src/util/featureSelection.cpp

```diff
@@ -29,12 +32,11 @@ uint32_t FeatureSelection::colorIdentifier(const Feature& _feature, const TileID
 
 bool FeatureSelection::clearFeaturesForTile(TileID _tileID) {
 
+    std::lock_guard<std::mutex> guard(m_mutex);
```

Comment:
Still the insertion into `m_tileFeatures` would not be synchronized with `find` then. Or `m_tileFeatures.end()` could be changed since `find` returned end.